lib/ansible/cli/galaxy.py

```diff
@@ -166,11 +222,19 @@ def add_delete_options(self, parser, parents=None):
         delete_parser.set_defaults(func=self.execute_delete)
 
     def add_list_options(self, parser, parents=None):
+        galaxy_type = 'role'
+        if parser.metavar == 'COLLECTION_ACTION':
+            galaxy_type = 'collection'
+
+        # Seprate methods for listing a role or collection
+        list_func = getattr(self, 'execute_list_%s' % galaxy_type)
+
         list_parser = parser.add_parser('list', parents=parents,
```

Comment:
This would fix the sanity error.
```suggestion
         list_parser = parser.add_parser('list_%s' % galaxy_type, parents=parents,
```
I wonder if hacking/build_library/build_ansible/command_plugins/generate_man.py should inspect parser.get_default(dest='func') instead/additionally, but maybe that's something to look into another time.

Reply:
I was looking at fixing it in `hacking/build_library/build_ansible/command_plugins/generate_man.py` because maybe we want to generate man pages for the collection options too? I really don't know exactly what that file is doing, so your fix is probably the better option for now.

Reply:
So this fixed the sanity test, but broke the CLI. I'm toying with some other ways to solve it.

Reply:
Ok, I think I got it sorted. I put `execute_list()` back in and it just calls the correct underlying method.